scripts/mapo

```diff
@@ -0,0 +1,231 @@
+    )
+    exec_args_group.add_argument(
+        "--sample-batch-size",
+        type=int,
+        default=10,
+        help="The size of the batch collected by workers (default=10).",
+    )
+    exec_args_group.add_argument(
+        "--train-batch-size",
+        type=int,
+        default=100,
+        help="The size of the batch for training (default=100).",
+    )
+
+    resources_args_group = parser.add_argument_group(">> Resources")
+    resources_args_group.add_argument(
+        "--num-workers",
+        type=int,
+        default=0,
+        help="The number of actors used for parallelism (default=0).",
+    )
+    resources_args_group.add_argument(
+        "--num-gpus",
+        type=float,
+        default=0.0,
+        help="The number of GPUs to allocate to the driver (default=0.0).",
+    )
+    resources_args_group.add_argument(
+        "--num-cpus",
+        type=int,
+        default=1,
+        help="The number of CPUs to allocate to the driver (default=1).",
+    )
+
+    exp_args_group = parser.add_argument_group(">> Experiment")
+    exp_args_group.add_argument(
+        "--num-samples",
+        type=int,
+        default=1,
+        help="The number of times to sample from the hyperparameter space "
+        "(default=1).",
+    )
+    exp_args_group.add_argument("--name", type=str, help="The name of the experiment.")
+
+    return parser.parse_args()
+
+
+def init():
+    """Initialize ray framework."""
+    ray.init()
+
+
+def register():
+    """Register all agents and custom environtments."""
+    register_all_agents()
+    register_all_environments()
+
+
+def run(args):
+    """Run experiments with configuration given by command-line arguments."""
+    trainer = args.run
+    experiment = args.name
+
+    analysis = tune.run(
+        trainer,
+        name=experiment,
+        verbose=args.verbose,
+        stop={"timesteps_total": args.timesteps_total},
+        # resources_per_trial={
+        #     "cpu": args.num_cpus,
+        #     "gpu": args.num_gpus,
+        # },
+        config={
+            "env": tune.grid_search(args.env),
+            "num_workers": args.num_workers,
+            "model": {
+                "custom_model": "mapo_model",
+                "custom_options": {
+                    "actor": args.config_actor_net,
+                    "critic": args.config_critic_net,
+                    "dynamics": args.config_dynamics_net,
+                },
+            },
+        },
+    )
+
+    return analysis
+
+
+def report(analysis):
+    """Report analysis for best configuraion in experiments."""
+    best_config = analysis.get_best_config("episode_reward_mean", mode="max")
+    best_logdir = analysis.get_best_logdir("episode_reward_mean", mode="max")
+
+    print(">> Best configuration:")
+    print(json.dumps(best_config, indent=2, sort_keys=True))
+    print()
+
+    print("tensorboard --logdir={}".format(best_logdir))
+    print()
+
```

Comment:
Looks pretty neat! I like the idea of automatically reporting the best results.

Reply:
Still need to learn how to deal with the [Analysis](https://ray.readthedocs.io/en/latest/tune-package-ref.html?highlight=ExperimentAnalysis#ray.tune.Analysis) object retuned by `ray.tune.run()` ...